data_structures/include/forehead/forehead.h

```diff
@@ -0,0 +1,76 @@
+#ifndef fore_head_h
+#define fore_head_h
+
+/**
+ * @file forehead.h
+ * @brief This header must be included in every sub-header file.
+ * @details
+ * 1. forehead.h included the necessary header files which needed by every
+ * sub-header file.
+ * 2. forehead.h defined some macros for debugging.
+ * @note These macro functions will be heavily used in the code (in situations
+ * such as exception handling, test cases, and termination functions)!
+ * @author danielhu19 (https://github.com/danielhu19)
+ */
+
+#include <assert.h>
+#include <limits.h>
+#include <stdbool.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <time.h>
+#include <unistd.h>
```

Comment:
Please add a one-line description of what the library/header is for (**see the example below**).

```c
#include <stdio.h>      /// for IO operations
#include <assert.h>    /// for assert
```